Add a contribution guideline for AlertmanagerConfig CRD. (#6897)
prometheus-operator/prometheus-operator: pkg/alertmanager/README.md

## Alertmanager Package

> The alertmanager package and CRDs together make up the alertmanager integration in the prometheus operator. Here is a

Reviewer: ```suggestion
The Alertmanager package and CRDs together make up the Alertmanager integration in the Prometheus-Operator. Here is a
```

Reviewer 2: Another way to say is that the pkg/alertmanager package implements the controller reconciling Alertmanager  and AlertmanagerConfig resources in a cluster.

quick overview of the overall package:

![pkg/alertmanager and crds](../../Documentation/img/am_readme.png)

### CRDs

Prometheus operator supports two CRDs related to Alertmanager.

> - Alertmanager CRD: Defines the alertmanager statefulset.

Reviewer: ```suggestion
- Alertmanager CRD: Deploys an Alertmanager instance in StatefulSet mode.
```

> - AlertmanagerConfig CRD: Deines the config for the alertmanager instances.

Reviewer: ```suggestion
- AlertmanagerConfig CRD: Defines the configuration for the Alertmanager instances.
```

The Alertmanager CRD is in [v1](https://github.com/prometheus-operator/prometheus-operator/blob/main/README.md#project-status).

> while AlertmanagerConfig CRD is currently in [beta](https://github.com/prometheus-operator/prometheus-operator/blob/main/README.md#project-status).

Reviewer 2: in practice we have 2 API versions for the AlertmanagerConfig CRD: v1alpha1 and v1beta1 (the former being the default stored version).

These CRDs are then picked up, validated and converted by the operator into configs.

### Updating AlertmanagerConfig CRD

If you are looking to make changes to the AlertmanagerConfig CRD, you would probably be making changes to the following files and for the reasons mentioned

>   <b>

Reviewer: Rather than <b></b> I would rather wrap the text with ** **

Reviewer: https://github.github.com/gfm/#emphasis-and-strong-emphasis

  pkg/apis/monitoring/v1alpa1/alertmanager_config_types.go 
  </b>

>   <br>

Reviewer: No need for `<br>`. Just add two spaces at the end of the line where you want a newline **mdox** generally adds an extra line break for `<br>` which can be avoided by this method.

Reviewer: This might help https://github.github.com/gfm/#example-660

   Update/Add/Delete the config in alpha.
   <br>

<br>

  <b>
   pkg/apis/monitoring/v1beta1/alertmanager_config_types.go 
  </b>

  <br>
   Update/Add/Delete the config in beta.
<br>

<br>

  <b>
   pkg/apis/monitoring/v1beta1/conversion_from.go
  </b>

  <br>
  Add conversion logic as Beta needs to be convertible from the hub version(alpha).

<br>

<br>

  <b>
  pkg/apis/monitoring/v1beta1/conversion_to.go
  </b>

  <br>
  Add conversion logic as Beta needs to be convertible to the hub version(alpha).

<br>

<br>

  <b>
  pkg/alertmanager/amcfg.go
  </b>

  <br>

>   CRD to config object conversion logic.

Reviewer: CRD to configure object conversion logic.

>   CRD to config object conversion logic.

Reviewer: ```suggestion
  CRD to configure object conversion logic.
```
